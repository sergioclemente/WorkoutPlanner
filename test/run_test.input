{name=Mixed units and percentage}
{output_unit=mi/hr}
(1km, 100), (1km, 5min/km)
-- EOT
(200m, 100)(45s, 100)
-- EOT
(2mi, 6:00min/mi, threshold)
-- EOT
(2mi, 80)
-- EOT
[(6min, 100), (2min, 80)], (10min, 70)
-- EOT
(2mi, 7:00min/mi)
-- EOT
(4s, 10mph), (5s, 6min/mi)
-- EOT
(45min, 3:44min/km)
-- EOT
(60min, 6:00min/mi)
-- EOT
(60min, 6:00 min/mi)
-- EOT
{output_unit=min/mi}
(10s, 60, relaxed)
-- EOT
(10s, relaxed)
-- EOT
{output_unit=/400meters}
(1mi, 100, threshold)
-- EOT