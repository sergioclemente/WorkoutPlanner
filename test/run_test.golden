Name: Mixed units and percentage
Input: 
(1km, 100), (1km, 5min/km)
Normalized: 
(1km, 100)
(1km, 74.6)
AST: 
ArrayInterval(
	SimpleInterval(1km, 100%, )
	SimpleInterval(1km, 5min/km, )
)
IF (Avg): 0.8635076320941238
TSS2: 11.3
Output Unit: mi/h
Pretty Print:
	* 1km @ 10mi/h
	* 1km @ 7.5mi/h
Duration Unit: km
Kilometers: 2
Duration (Sec): 523.6936292054402
----------------------------
Input: 
(200m, 100)(45s, 100)
Normalized: 
(200m, 100)
(45sec, 100)
AST: 
ArrayInterval(
	SimpleInterval(200m, 100%, )
	SimpleInterval(45sec, 100%, )
)
IF (Avg): 1
TSS2: 2.5
Dominant Unit: %
Pretty Print:
	* 200m @ 100%
	* 45'' @ 100%
Duration (Sec): 89.73872584108804
----------------------------
Input: 
(2mi, 6:00min/mi, threshold)
Normalized: 
(2mi, 100, threshold)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 6:00min/mi, threshold)
)
IF (Avg): 1
TSS2: 20
Dominant Unit: min/mi
Pretty Print:
	* 2mi threshold @ 6:00min/mi
Duration Unit: mi
Miles: 2mi
Duration (Sec): 720
----------------------------
Input: 
(2mi, 80)
Normalized: 
(2mi, 80)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 80%, )
)
IF (Avg): 0.8
TSS2: 16
Dominant Unit: %
Pretty Print:
	* 2mi @ 80%
Duration Unit: mi
Miles: 2mi
Duration (Sec): 900
----------------------------
Input: 
[(6min, 100), (2min, 80)], (10min, 70)
Normalized: 
[(6min, 100), (2min, 80)]
(10min, 70)
AST: 
ArrayInterval(
	ArrayInterval(
		SimpleInterval(6min, 100%, )
		SimpleInterval(2min, 80%, )
	)
	SimpleInterval(10min, 70%, )
)
IF (Avg): 0.8225975119502044
TSS2: 19.7
Dominant Unit: %
Pretty Print:
	* 6' @ 100% - 2' @ 80%
	* 10' @ 70%
Duration Unit: min
Miles: 2.4mi
Duration (Sec): 1080
----------------------------
Input: 
(2mi, 7:00min/mi)
Normalized: 
(2mi, 85.7)
AST: 
ArrayInterval(
	SimpleInterval(2mi, 7:00min/mi, )
)
IF (Avg): 0.8571428571428571
TSS2: 18
Dominant Unit: min/mi
Pretty Print:
	* 2mi @ 7:00min/mi
Duration Unit: mi
Miles: 2mi
Duration (Sec): 840
----------------------------
Input: 
(4s, 10mph), (5s, 6min/mi)
Normalized: 
(4sec, 100)
(5sec, 100)
AST: 
ArrayInterval(
	SimpleInterval(4sec, 10mi/h, )
	SimpleInterval(5sec, 6:00min/mi, )
)
IF (Avg): 1
TSS2: 0.3
Duration Unit: s
Duration (Sec): 9
----------------------------
Input: 
(45min, 3:44min/km)
Normalized: 
(45min, 99.9)
AST: 
ArrayInterval(
	SimpleInterval(45min, 3.7min/km, )
)
IF (Avg): 0.9986322732385723
TSS2: 74.9
Dominant Unit: min/km
Pretty Print:
	* 45' @ 3:44min/km
Duration Unit: min
Kilometers: 12.1km
Duration (Sec): 2700
----------------------------
Input: 
(60min, 6:00min/mi)
Normalized: 
(1hr, 100)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 6:00min/mi, )
)
IF (Avg): 1
TSS2: 100
Dominant Unit: min/mi
Pretty Print:
	* 1hr @ 6:00min/mi
Duration Unit: min
Miles: 10mi
Duration (Sec): 3600
----------------------------
Input: 
(60min, 6:00 min/mi)
Normalized: 
(1hr, 100)
AST: 
ArrayInterval(
	SimpleInterval(1hr, 6:00min/mi, )
)
IF (Avg): 1
TSS2: 100
Dominant Unit: min/mi
Pretty Print:
	* 1hr @ 6:00min/mi
Duration Unit: min
Miles: 10mi
Duration (Sec): 3600
----------------------------
Input: 
(10s, 60, relaxed)
Normalized: 
(10sec, 60, relaxed)
AST: 
ArrayInterval(
	SimpleInterval(10sec, 60%, relaxed)
)
IF (Avg): 0.6
TSS2: 0.1
Output Unit: min/mi
Pretty Print:
	* 10'' relaxed @ 10:00min/mi
Duration Unit: s
Miles: 0mi
Duration (Sec): 10
----------------------------
Input: 
(10s, relaxed)
Normalized: 
(10sec, relaxed)
AST: 
ArrayInterval(
	SimpleInterval(10sec, , relaxed)
)
IF (Avg): 0.01
TSS2: 0
Dominant Unit: %
Pretty Print:
	* 10'' relaxed
Duration Unit: s
Miles: 0mi
Duration (Sec): 10
----------------------------
Input: 
(1mi, 100, threshold)
-- EOTNormalized: 
(1mi, 100, threshold)
AST: 
ArrayInterval(
	SimpleInterval(1mi, 100%, threshold)
)
IF (Avg): 1
TSS2: 10
Output Unit: /400meters
Pretty Print:
	* 1mi threshold @ 1:29/400meters
Duration Unit: mi
Miles: 1mi
Duration (Sec): 360
----------------------------
